src/code-view.js

```diff
@@ -14,70 +14,102 @@
+            $fullscreenButton = $("<span class='fs-button-container' title='Expand to fullscreen'></span>");
             $fullscreenButton.on('click', function () { window.open($iframe.attr("src") || $iframe.attr("data-src"))});
             $fullscreenButton.appendTo($navbar);
 
             //Render the code view widget
             $(this.element).prepend($navbar);
 
-            // Render a footer with CSB and SB buttons (if any !!) 
-            this._renderFooter($footer, this.element);
+            this._options = {
+              $navbar: $navbar,
+              $codeViewsContainer: $codeViewsContainer,
+              $activeTab: $exampleTab,
+              $activeView: $sampleContainer,
+              $footer: $footer
+            }
 
         },
         _codeViewTabClick: function(event) {
             var $tab = $(event.target);
             if(!$tab.hasClass("." + this.css.tab + "--active")){
-                $('#cv-' + this.options.iframeId + " .code-view-tab--active").switchClass(this.css.tab + "--active", this.css.tab,  0);
+                this._options.$activeTab.switchClass(this.css.tab + "--active", this.css.tab,  0)
                 $tab.switchClass(this.css.tab, this.css.tab + "--active", 0);
+                this._options.$activeTab = $tab;
                 $tab.is('[tab-content-id=' + 'code-view-' + this.options.iframeId + '-' + 'example-tab-content]') ? $tab.siblings('.fs-button-container').css('visibility', 'visible') :
                                                                                                                     $tab.siblings('.fs-button-container').css('visibility', 'hidden')
-                $('#cv-' + this.options.iframeId + ' > .' + this.css.viewContainer + ' > .'+ this.css.tabContent).css('display', 'none');
-                $('#' + $tab.attr('tab-content-id')).css('display', 'block');
+                this._options.$activeView.css('display', 'none');
+                this._options.$activeView = $('#' + $tab.attr('tab-content-id'))
+                this._options.$activeView.css('display', 'block');
             }
         },
-        _createTabsWithCodeViews: function($navbar, $codeViewsContainer) {
+        createTabsWithCodeViews: function(filesData) {
             var self = this;
-
+            if(!filesData || filesData.length === 0){
+              return;
+            }
             var isEmptyFile = new RegExp('^[ \t\r\n]*$');
-            this.options.files.filter(function (f)  {return !isEmptyFile.test(f.content)}).forEach(function (f){
-                var language = f.fileExtension === 'ts' ? 'typescript' : f.fileExtension;
+
+            var _filesData = filesData.filter(function (f)  {return !isEmptyFile.test(f.content)});
+
+            if(_filesData.length === 0) {
+              return;
+            }
+            
+            this.options.files = _filesData;
+            var headers = _filesData.map(f => f.fileHeader);
```

Comment:
This arrow function is not compatible with IE and leads to syntax error and the sample is not rendered.